spring-amqp/src/main/java/org/springframework/amqp/core/AnonymousQueue.java

```diff
@@ -64,17 +67,42 @@ public AnonymousQueue(NamingStrategy namingStrategy) {
 	 * supplied arguments.
 	 * @param namingStrategy the naming strategy.
 	 * @param arguments the arguments.
+	 * @deprecated in favor of {@link #AnonymousQueue(NamingStrategy, Map)}.
+	 *
 	 * @since 1.5.3
 	 */
+	@Deprecated
 	public AnonymousQueue(NamingStrategy namingStrategy, Map<String, Object> arguments) {
 		super(namingStrategy.generateName(), false, true, true, arguments);
 	}
 
+	/**
+	 * Construct a queue with a name provided by the supplied naming strategy.
+	 * @param namingStrategy the naming strategy.
+	 * @since 2.1
+	 */
+	public AnonymousQueue(org.springframework.amqp.core.NamingStrategy namingStrategy) {
+		this(namingStrategy, null);
+	}
+
+	/**
+	 * Construct a queue with a name provided by the supplied naming strategy with the
+	 * supplied arguments.
+	 * @param namingStrategy the naming strategy.
+	 * @param arguments the arguments.
+	 * @since 2.1
+	 */
+	public AnonymousQueue(org.springframework.amqp.core.NamingStrategy namingStrategy, Map<String, Object> arguments) {
+		super(namingStrategy.generateName(), false, true, true, arguments);
+	}
+
 	/**
 	 * A strategy to name anonymous queues.
+	 * @deprecated - use the {@link org.springframework.amqp.core.NamingStrategy}.
 	 * @since 1.5.3
 	 *
 	 */
+	@Deprecated
 	@FunctionalInterface
 	public interface NamingStrategy {
```

Comment:
Can't we make this as `extends org.springframework.amqp.core.NamingStrategy` for more smooth migration in the future?